galaxy/api/views/namespace.py

```diff
@@ -48,9 +48,13 @@
 def check_basic(data, errors):
     if not data.get('name'):
         errors['name'] = "Attribute 'name' is required"
-    elif not re.match('^[\w-]+$', data['name']):
-        # Allow only names containing word chars and '-'
-        errors['name'] = "Name can only contain [A-Za-z0-9-_]"
+    elif not re.match('^[\w]+$', data.get('name')):
+        # Allow only names containing word chars
+        errors['name'] = "Name can only contain [A-Za-z0-9_]"
+    elif(len(data.get('name')) <= 2):
```

Comment:
Possible `TypeError` here. You're using `data.get('name')` multiple times.
Use `name = data.get('name')` in the beginning of the function and work with a variable below.